sites/website/src/docs/resources/cheat-sheet.md

```diff
@@ -0,0 +1,711 @@
+| HTML Attribute | `<a href=${…}></a>` | Creates a binding that uses the setAttribute API. Attribute bindings also support interpolation with text and other bindings. |
+| HTML Boolean Attribute | `<input ?disabled=${…}>` | Creates a binding that adds or removes the attribute based on truthy/falsey values. |
+| JS Property | `<input :value=${…}>` | Creates a binding that sets a JavaScript property on the element. |
+| Event Handler | `<button @click=${…}>...</button>` | Registers an event handler using addEventListener. The listener is automatically removed when the template is unbound. |
+| HTML Element Reference | `<button ${ref('myButton')}>...</button>` | Captures a reference to the element and assigns it to the named property on the data source. |
+| Slotted Node Capture | `<slot ${slotted('defaultSlotNodes')}></slot>` | Watches the slot for changes and synchronizes those to an array, assigned to the named property on the data source. |
+| Child Node Capture | `<div ${children('divChildren')}></div>` | Watches the element's children or changes and synchronizes those to an array, assigned to the named property on the data source. |
+
+---
+### [Directives](https://www.fast.design/docs/fast-element/using-directives)
+
+Use the [`when`](https://www.fast.design/docs/fast-element/using-directives#the-when-directive) directive to conditionally render blocks of HTML:
+
+```ts
+import { FASTElement, customElement, observable, html, when } from '@microsoft/fast-element';
+
+const template = html<MyApp>`
+ ...
+  ${when(x => !x.ready, html<MyApp>`
+    Loading...
+  `)}
+`;
+
+@customElement({
+  name: 'my-app',
+  template
+})
+export class MyApp extends FASTElement {
+  @observable ready: boolean = false;
+  ...
+}
+```
+
+---
+
+Use the [`repeat`](https://www.fast.design/docs/fast-element/using-directives#the-repeat-directive) directive to render a list of data:
+
+
+```ts
+import { FASTElement, customElement, observable, html, repeat } from '@microsoft/fast-element';
+
+const template = html<FriendList>`
+  ...
+    ${repeat(x => x.friends, html<string>`
+      <li>${x => x}</li>
+    `)}
+`;
+
+@customElement({
+  name: 'friend-list',
+  template
+})
+export class FriendList extends FASTElement {
+  @observable friends: Person[] = [];
+  ...
+}
+```
+
+---
+
+Properties available on the context object within a `repeat` block:
+
+| Property | Definition |
+| :-- | :-- |
+| event | The event object when inside an event handler. |
+| parent | The parent view-model when inside a repeat block. |
+| parentContext | The parent ExecutionContext when inside a repeat block. This is useful when repeats are nested and the inner-most repeat needs access to the root view-model. |
+| index | The index of the current item when inside a repeat block (opt-in). |
+| length | The length of the array when inside a repeat block (opt-in). |
+| isEven | True if the index of the current item is even when inside a repeat block (opt-in). |
```

Comment:
I think we should break apart the `opt-in` properties here. It's been a point of confusion in the past because those properties don't exist sometimes.

If we break this out into two tables, we can identify which properties will *always* exist and which properties will only exist when providing the position configuration. We also show an example of *how* to provide the positioning configuration:

```html
<ul>
  ${repeat(x => x.friends, html<string>`
    <li>${(x, c) => c.index} ${x => x}</li>
  `, { positioning: true })}
</ul>
```

Reply:
@nicholasrice Great points, thank you - updated!

Reply:
Would you mind checking my updates on this ^ to see if they make sense?

Reply:
Sorry for the delay, looks great! thank you!